ntropy_sdk/recurring_payments.py

```diff
@@ -0,0 +1,227 @@
+    is_essential: bool
+    is_active: bool
+    first_payment_date: str
+    latest_payment_date: str
+    next_expected_payment_date: Optional[str]
+    latest_payment_description: str
+    transaction_ids: List[Union[int, str]]
+    transactions: List[Any]  # List[EnrichedTransaction]
+    total_amount: float
+    iso_currency_code: Optional[str]
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]):
+        return cls(
+            periodicity=data.get("periodicity"),
+            latest_payment_amount=data.get(
+                "latest_payment_amount", data.get("amount", 0)
+            ),
+            merchant=data.get("merchant"),
+            website=data.get("website"),
+            logo=data.get("website"),
+            labels=data.get("labels", []),
+            type=data.get("type"),
+            is_essential=data.get("is_essential"),
+            is_active=data.get("is_active"),
+            first_payment_date=data.get("first_payment_date"),
+            latest_payment_date=data.get("latest_payment_date"),
+            next_expected_payment_date=data.get("next_expected_payment_date"),
+            latest_payment_description=data.get("latest_payment_description", ""),
+            transaction_ids=data.get("transaction_ids", []),
+            transactions=data.get("transactions", []),
+            total_amount=data.get("total_amount", 0),
+            iso_currency_code=data.get("iso_currency_code"),
+        )
+
+    def to_df(self) -> Any:
+        try:
+            import pandas as pd
+        except ImportError:
+            raise RuntimeError("pandas is not installed")
+        return pd.DataFrame(self.json())
+
+    def _repr_df(self) -> Any:
+        try:
+            import pandas as pd
+        except ImportError:
+            raise RuntimeError("pandas is not installed")
+        df = self.to_df()
+        if df.empty:
+            return df
+        df = df.fillna("N/A")
+        df.transaction_ids = df.transaction_ids.apply(lambda x: len(x))
+        df = df.rename({"transaction_ids": "# transactions"})
+        return df
+
+    def _repr_html_(self) -> Union[str, None]:
+        # used by ipython/jupyter to render
+        try:
+            import pandas as pd
+
+            df = self._repr_df()
+            if df.empty:
+                return f"{self.__class__.__name__}([])"
+            return df._repr_html_()
+        except ImportError:
+            # pandas not installed
+            return self.__repr__()
+
+    def __repr__(self) -> str:
+        try:
+            import pandas as pd
+
+            df = self._repr_df()
+            if df.empty:
+                return f"{self.__class__.__name__}([])"
+            return tabulate(df, showindex=False)
+        except ImportError:
+            # pandas not installed
+            repr = str([ig.dict(exclude={"transaction_ids"}) for ig in self.list])
+            return f"{self.__class__.__name__}({repr})"
+
+
+class RecurringPaymentsGroups(list):
```

Comment:
same comments as above for list

Reply:
len(recurring_payments_groups) works so does recurring_payments_groups[0], not sure about add